src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -185,6 +189,18 @@ private StatementSyntax Transform( StatementSyntax stmt )
 			_ => UnhandledSyntax( stmt )
 		};
 
+	private StatementSyntax Transform( ReturnStatementSyntax returnStmt ) {
+		var expr = returnStmt.Expression;
+		if( m_generatedFunctionReturnsVoid && expr is not null ) {
+			if( expr is InvocationExpressionSyntax || expr is AssignmentExpressionSyntax || expr is IIncrementOrDecrementOperation || expr is DeclarationExpressionSyntax || expr is MemberAccessExpressionSyntax ) {

```

Comment:
Nice -- this could maybe be a helper, and you could use `.RawKind` to do a switch (better perf than reflection/the `||`s (maybe on that second bit)):

```cs
private static bool IsStatementCompatibleExpression( ExpressionSyntax expr ) {
  switch( expr.RawKind ) {
    case SyntaxKind.InvocationExpression:
    case SyntaxKind.AssignmentExpression:
    // ...
      return true;
    default:
      return false;
  }
}
```

something like that

Reply:
I would rather not use Kind because there is no SyntaxKind.AssignmentExpression, I would have to add cases for all of these:
![image](https://github.com/Brightspace/D2L.CodeStyle/assets/20260142/572b3a89-9ab0-4322-bd48-b4e48802d79e)
Could I do something like this?
```
private static bool IsStatementCompatibleExpression( ExpressionSyntax expr ) {
  switch( expr) {
    case InvocationExpressionSyntax e:
    case AssignmentExpressionSyntax e:
    // ...
      return true;
    default:
      return false;
  }
}
```

Reply:
Ah ok. A helper would still be good IMO just to keep things tidy but you don't have to do it in this PR.